### ISSUE TITLE
ScreenRay bug(not sure)!

### ISSUE THREAD

**Issue by chenjie199234** [urho3d/Urho3D#2357]
My PC:Win10 Version:1803
Graphics:DX11
CameraType:Orthographic
i didn't move camera~~~
didn't do anything
just put my mouse on somewhere and click my mouse.
i output the ray's attribute both in E_MOUSEBUTTONUP and E_UPDATE
E_MOUSEBUTTONUP has the right result!
here is the result!
![log](https://user-images.githubusercontent.com/29348971/43673019-faa8f664-97ed-11e8-8162-d89d5b4d24bc.PNG)
here is the code
![code](https://user-images.githubusercontent.com/29348971/43673023-059287ac-97ee-11e8-9f25-6bb7b0b05c15.PNG)

+++++++++++++++++++++++++++++++++++
I found which code causes this problem!
If i use this api in the E_UPDATE callback,then this problem will show!
API is SetOrthoSize in Camera Class.
I use it in E_UPDATE like this:
game_camera_node_->GetComponent&lt;Camera&gt;()->SetOrthoSize(s);
I will change the s with my mouse wheel in W_MOUSEWHEEL callback,and update it in E_UPDATE callback with this API,so that i can zoom in and out when the camera type is Orthographic.
This API in E_UPDATE callback caused this problem,I don't know if this is a bug!
when i put this API in E_MOUSEWHEEL callback,problem gone!

**Comment by dertom95** [comment]
@chenjie199234 
I had the same problem. Also orthocam, also using camera's screenray and also changing the OrthoSize on mousewheel-event. 
When setting the othoSize directly on the event everything was fine. Then I changed the mechanism to smooth the OrthoSize by lerping to a targetScale inside the Update-Method and mousepicking stopped working. If found out that there was some strange offset to the ray when the camera was moved/updated in the same frame and since I was lerping all the time that behaviour occured also all the time.
Once I added to stop camera-orthoScale-updates when the targetScale is almost met, the result was correct again.
